rust/src/import_parsing.rs

```diff
@@ -0,0 +1,595 @@
+use crate::errors::GrimpResult;
+use ruff_python_ast::statement_visitor::{walk_body, walk_stmt, StatementVisitor};
+use ruff_python_ast::{Expr, Stmt};
+use ruff_python_parser::parse_module;
+use ruff_source_file::{LineIndex, SourceCode};
+use std::fs;
+use std::path::Path;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ImportedObject {
+    pub name: String,
+    pub line_number: usize,
+    pub line_contents: String,
+    pub typechecking_only: bool,
```

Comment:
One optimization we could consider for the future: we don't actually need to keep track of whether an import is within a type checking guard here (it's deliberately not exposed in Grimp's public API). Instead, we could just take `exclude_type_checking_imports` as an argument and use a (I'm guessing) slightly faster algorithm if it's false.